jep/0000/README.adoc

```diff
@@ -0,0 +1,355 @@
+* link:https://jenkins.io/blog/2018/06/17/running-jenkins-with-java10-11/[“Running Jenkins with Java 10 and 11”] blogpost is updated to refer the new guidelines
+* For Java 11 the website should be updated only after the official release of OpenJDK 11
+* There is an announcement blogpost for Java 10 support general availability in weekly
+** The blogpost will include upgrade guidelines, “make a backup” will one of the required steps there
+* There is an announcement blogpost for Java 10 support general availability in LTS
+
+==== Wiki
+
+* There is a Wiki page created to track known Java 10+ incompatibilities in the Jenkins Core and Plugins.
+* The page will have format similar to link:https://wiki.jenkins.io/display/JENKINS/Plugins+affected+by+fix+for+JEP-200[Plugins affected by fix for JEP-200] Wiki
+
+==== Post-release support
+
+After the release of Java 10+ support, there may be a number of defects created by early adopters.
+It may cause additional workload on plugin and core maintainers, and this JEP sets sets a requirement to assist with triaging of issues after the release.
+
+After the weekly release availability the JEP sponsor (or a group of people nominated by him, “Java 10+ Maintainers”) will be responsible to provide an extra support for the issues:
+
+* Java 10+ Maintainers will periodically review open defects and triage them (e.g. once per week)
+* Java 10+ Maintainers may request additional information from the reporter. Finally, they are expected to communicate the triage outcome.
+* Possible triage outcomes:
+** Accepted by Java 10+ Maintainers. In such case one of maintainers assigns the issue to himself and delivers the fix
+** Rejected by Java 10+ Maintainers - functional defect in the plugin (e.g. reliance on Java version or private fields in Reflections) or lack of justification for a fix
+** Issue is closed - Not a defect, Duplicate, etc.
+* For accepted issues maintainers will prioritize and schedule the fix
+** Java 10 support is considered as a “Feature” with an obvious workaround: “Downgrade to Java 8”
+** Fixes for Java 10 will be prioritized by the team, but incompatibilities won’t be considered as Blocker issues if downgrade is possible
+* Issues rejected by Java 10+ maintainers will be assigned to component leads in JIRA (if any).
+
+The proposed support model will be in place until “Availability in LTS + 2 months”.
+After this period Jenkins component maintainers will be responsible for triaging and fixing issues in their components.
+SECURITY reports will be triaged by Jenkins Security Team.
+
+
+== Motivation
+
+Non-goals in the specification are defined to limit the scope of work.
+The main objective is to get Jenkins running with Java 10+,
+there will be follow-up tasks to cleanup Illegal Reflective Access warnings and to adopt new features.
+
+== Reasoning
+
+Reasoning for the Specification decisions is provided in the “Approach” section above.
```

Comment:
I don't see an "approach" section, but also reasoning goes here not somewhere else.

Reply:
I renamed it to "Goals and non-goals" in specification.
IMHO it fits better as a part of specification, not Reasoning